platform.txt

```diff
@@ -4,10 +4,9 @@ version=2.0.0
 runtime.tools.xtensa-esp32-elf-gcc.path={runtime.platform.path}/tools/xtensa-esp32-elf
 runtime.tools.xtensa-esp32s2-elf-gcc.path={runtime.platform.path}/tools/xtensa-esp32s2-elf
 
-tools.esptool_py.path={runtime.platform.path}/tools/esptool
```

Comment:
the reason why esptool is defined this way is because the path gets replaced when packaging for the board manager. Can not accept this in the main repo :) it will break the process.

Reply:
ah, sorry, I didn't know about that. is trying to remove duplication code, maybe we could have an variable for upload e.g

```
tools.esptool_py.upload.pattern_args=--chip {build.mcu} --port "{serial.port}" --baud {upload.speed} {upload.flags} --before default_reset --after hard_reset write_flash -z --flash_mode {build.flash_mode} --flash_freq {build.flash_freq} --flash_size detect 0xe000 "{runtime.platform.path}/tools/partitions/boot_app0.bin" 0x1000 "{build.path}/{build.project_name}.bootloader.bin" 0x10000 "{build.path}/{build.project_name}.bin" 0x8000 "{build.path}/{build.project_name}.partitions.bin"

tools.esptool_py.upload.pattern="{path}/{cmd}" {pattern_args}
tools.esptool_py.upload.pattern.linux=python "{path}/{cmd}" {pattern_args}
```
Though it is totally up to you, I would revert it if that is your preference, since you are the maintainer.

Reply:
sure :) any optimisation and removing of duplication is great :) as long as that line stays the same so it will be properly replaced